### PR TITLE
RUn gitleaks as pull_request_target

### PR DESCRIPTION
**Pull request by KacperMalachowski**
**Description**

`pull_request_target` triggered workflows are run in context of base branch.

Changes proposed in this pull request:

- Use pull request target as the most interesting part is commits in base branch

**Related issue(s)**

/kind bug
/area autoamtion

### CODE REVIEW

File: .ci-config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (Sawthis) [inline .ci-config.json:1]: I think when you use pull_request_target this file is not needed and vars can be provided as here: https://github.com/kyma-project/test-infra/blob/main/.github/workflows/pull-plan-prod-terraform.yaml#L47-L48

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-gitleaks.yml [hunk 2]
```diff
@@ -1,8 +1,9 @@
 name: pull-gitleaks
 on:
-  pull_request:
+  pull_request_target:
```
Comment (Sawthis) [inline .github/workflows/pull-gitleaks.yml:3]: Additionally with target you need to adjust the checkout to https://github.com/kyma-project/test-infra/pull/10153/files#diff-cb4389f8b525d78c165731a8d28493962902fc9617910342d239fb01093d9081R17 or https://github.com/kyma-project/test-infra/blob/main/.github/workflows/pull-plan-prod-terraform.yaml#L30

**Review by Sawthis — CHANGES_REQUESTED** [review]

File: .ci-config.json (hunk 1 above)
Comment (KacperMalachowski) [reply .ci-config.json:1]: There're was a discussion with @dekiel to stick for it now and discuss it further in the team.

**Review by Sawthis — APPROVED** [review]

**Pull request merged**